README.md

```diff
@@ -17,16 +17,112 @@ pip install -r requirements.txt
 pre-commit install
 ```
 ## linux
-make sure to have pip
+use the `make` command with an `action` when developing on linux.  think of `actions` as a list of predefined commands to help simplify common development routines.
+
+#### view avaiable actions
+`make help` to see all avaiable actions
+
 ```sh
-sudo apt install python3-pip
+public-api$ make help
+clean-pyc            clean python cache files
+clean-test           cleanup pytests leftovers
+test                 Run pytest unit tests
+test-debug           Run unit tests with debugging enabled
+test-coverage        Run unit tests and check code coverage
+docker-up            Startup docker
+docker-build         Startup docker
+setup                setup & run after downloaded repo
+pre-commit-setup     Install pre-commit
+pre-commit           Run pre-commit
 ```
+
+### setup your enviornment
+prereqs: 
+- docker
+- python3
+- local clone of repo
+- terminal opened to cloned repos root path
+
+run `setup`
 ```sh
-python3 -m venv .venv

```

Comment:
i jus noticed none of this is running in a venv? either we tell them to create a venv or we create a venv?
e.g python -m venv .venv

Reply:
yea we can put it in a venv.  i dont like managing them when i end up having multiple virtual environments for python.  little bit of overhead to figure out what venvs are available, what project is using which and cleaning up old ones that you dont use.  

if we do maybe we should name it project specific venv name.  like venv-public-api.

Reply:
this should be resolved now, added to read me step to create venv and activate it